### PR TITLE
(DOCSP-14643): Port Swift FAQ info from Realm.io

### PR DESCRIPTION
**Pull request by dacharyc**
## Pull Request Info

From the legacy doc [Swift: FAQ](https://docs.mongodb.com/realm-legacy/docs/swift/latest/index.html#faq)

I did not port content for:
- Is Realm open source?
- I see a network call to Mixpanel when I run my app

The first question seemed more like a top-level Realm docs question than a Realm Swift question, and I'm not sure it warrants a FAQ. 

The second bullet also seems to apply to more than just the Swift SDK, and I was uncertain how/where we'd want to comment on this, if at all. The example linked to in the ["source code" link](https://github.com/realm/realm-cocoa/blob/master/Realm/RLMAnalytics.mm) is the Realm Cocoa implementation of analytics, but I found stuff in several of the SDK repos related to this, so I think it applies more broadly.

And finally, I added info on finding the Realm file path to the Realm Studio page for Swift and Objective-C, but if we want to keep this there, we should probably make a ticket to add corresponding info for the other SDKs/frameworks.

### Jira

- https://jira.mongodb.org/browse/DOCSP-14643

### Staged Changes (Requires MongoDB Corp SSO)

- [Fundamentals -> Object Models & Schemas -> Realm Does Not Support Swift Structs](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-14643/sdk/ios/fundamentals/object-models-and-schemas/#realm-does-not-support-swift-structs)
- [Fundamentals -> Realms -> App Download File Size](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-14643/sdk/ios/fundamentals/realms/#app-download-file-size)
- [Realm Studio -> Find the Realm File(s)](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-14643/sdk/ios/fundamentals/realms/#app-download-file-size)

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/sdk/ios/fundamentals/object-models-and-schemas.txt [hunk 1]
```diff
@@ -109,6 +109,37 @@ boilerplate query code and capacity for error.
    To learn how to add relationships to your Realm object, see
    :ref:`ios-declare-relationship-properties`.
 
+Realm Does Not Support Swift Structs
+------------------------------------
+
+{+client-database+} does not currently support structs as models for a 
+variety of reasons. Foremost, {+service-short+}'s design focuses on
```
Comment (cbush) [inline source/sdk/ios/fundamentals/object-models-and-schemas.txt:116]: Word choice:
```suggestion
variety of reasons. {+service-short+}'s design focuses on 
```

File: source/sdk/ios/fundamentals/object-models-and-schemas.txt [hunk 2]
```diff
@@ -109,6 +109,37 @@ boilerplate query code and capacity for error.
    To learn how to add relationships to your Realm object, see
    :ref:`ios-declare-relationship-properties`.
 
+Realm Does Not Support Swift Structs
+------------------------------------
+
+{+client-database+} does not currently support structs as models for a 
+variety of reasons. Foremost, {+service-short+}'s design focuses on 
+“live” objects. This concept is not compatible with value type structs. 
+By design, {+service-short+} provides features which are incompatible with
```
Comment (cbush) [inline source/sdk/ios/fundamentals/object-models-and-schemas.txt:118]:
```suggestion
By design, {+service-short+} provides features that are incompatible with 
```

File: source/sdk/ios/fundamentals/object-models-and-schemas.txt [hunk 3]
```diff
@@ -109,6 +109,37 @@ boilerplate query code and capacity for error.
...
+Realm Does Not Support Swift Structs
+------------------------------------
+
+{+client-database+} does not currently support structs as models for a 
+variety of reasons. Foremost, {+service-short+}'s design focuses on 
+“live” objects. This concept is not compatible with value type structs. 
+By design, {+service-short+} provides features which are incompatible with 
+these semantics, such as:
+ 
+- Live data
```
Comment (cbush) [inline source/sdk/ios/fundamentals/object-models-and-schemas.txt:121]: Link to live objects page?

File: source/sdk/ios/fundamentals/object-models-and-schemas.txt [hunk 4]
```diff
@@ -109,6 +109,37 @@ boilerplate query code and capacity for error.
...
+- Reactive APIs
```
Comment (cbush) [inline source/sdk/ios/fundamentals/object-models-and-schemas.txt:122]: Link to react to changes page?

File: source/sdk/ios/fundamentals/object-models-and-schemas.txt [hunk 5]
```diff
@@ -109,6 +109,37 @@ boilerplate query code and capacity for error.
...
+- Low memory footprint of data
+- Good operation performance
+- Lazy and cheap access to partial data
```
Comment (cbush) [inline source/sdk/ios/fundamentals/object-models-and-schemas.txt:125]: Live queries link?

File: source/sdk/ios/fundamentals/object-models-and-schemas.txt [hunk 6]
```diff
@@ -109,6 +109,37 @@ boilerplate query code and capacity for error.
...
+- Lack of data serialization/deserialization
+- Keeping potentially complex object graphs synchronized
```
Comment (cbush) [inline source/sdk/ios/fundamentals/object-models-and-schemas.txt:127]: Link to Sync?

File: source/sdk/ios/fundamentals/object-models-and-schemas.txt [hunk 7]
```diff
@@ -109,6 +109,37 @@ boilerplate query code and capacity for error.
...
+“live” objects. This concept is not compatible with value type structs. 
+By design, {+service-short+} provides features which are incompatible with 
+these semantics, such as:
+ 
+- Live data
+- Reactive APIs
+- Low memory footprint of data
+- Good operation performance
+- Lazy and cheap access to partial data
+- Lack of data serialization/deserialization
```
Comment (cbush) [inline source/sdk/ios/fundamentals/object-models-and-schemas.txt:126]: Link to Object Models & Schemas?

File: source/sdk/ios/fundamentals/object-models-and-schemas.txt [hunk 8]
```diff
@@ -109,6 +109,37 @@ boilerplate query code and capacity for error.
...
+- Keeping potentially complex object graphs synchronized
+
+That said, it is sometimes useful to detach objects from their backing 
+{+realm+}. This typically isn't an ideal design decision. Instead, 
+developers use this as a workaround for temporary limitations in our 
+library. 
+
+To make this easier, we’ve implemented standalone/detached {+service-short+} 
+objects to behave like :apple:`Apple's NSObject <documentation/objectivec/nsobject>`.
```
Comment (cbush) [inline source/sdk/ios/fundamentals/object-models-and-schemas.txt:135]: "Implemented standalone/detached ... to behave" reads a bit weird. How about:

>We've made unmanaged objects behave like ..."

File: source/sdk/ios/fundamentals/object-models-and-schemas.txt [hunk 9]
```diff
@@ -109,6 +109,37 @@ boilerplate query code and capacity for error.
...
+- Keeping potentially complex object graphs synchronized
+
+That said, it is sometimes useful to detach objects from their backing 
+{+realm+}. This typically isn't an ideal design decision. Instead, 
+developers use this as a workaround for temporary limitations in our 
+library. 
+
+To make this easier, we’ve implemented standalone/detached {+service-short+} 
+objects to behave like :apple:`Apple's NSObject <documentation/objectivec/nsobject>`. 
+We support making “in-memory copies” of {+service-short+} objects. Use an initializer
```
Comment (cbush) [inline source/sdk/ios/fundamentals/object-models-and-schemas.txt:136]: Watch out for smart quotes:
```suggestion
We support making "in-memory copies" of {+service-short+} objects. Use an initializer 
```

File: source/sdk/ios/fundamentals/object-models-and-schemas.txt [hunk 10]
```diff
@@ -109,6 +109,37 @@ boilerplate query code and capacity for error.
...
+- Lazy and cheap access to partial data
+- Lack of data serialization/deserialization
+- Keeping potentially complex object graphs synchronized
+
+That said, it is sometimes useful to detach objects from their backing 
+{+realm+}. This typically isn't an ideal design decision. Instead, 
+developers use this as a workaround for temporary limitations in our 
+library. 
+
+To make this easier, we’ve implemented standalone/detached {+service-short+}
```
Comment (cbush) [inline source/sdk/ios/fundamentals/object-models-and-schemas.txt:134]:
```suggestion
To make this easier, we've implemented standalone/detached {+service-short+} 
```

File: source/sdk/ios/fundamentals/object-models-and-schemas.txt [hunk 11]
```diff
@@ -121,3 +152,6 @@ Summary
 
 - You can define to-one, to-many, and inverse relationships in your schema.
   Inverse relationships automatically form backlinks.
+
+- {+service-short+} does not currently support structs. When required, 
+  create "in-memory copies" of {+service-short+} objects.
```
Comment (cbush) [inline source/sdk/ios/fundamentals/object-models-and-schemas.txt:157]: Does this make it sound more complicated than it is? I feel like the point here is that you can use "unmanaged" objects like normal objects. (Standalone, in-memory, detached are all new words for "unmanaged")

File: source/sdk/ios/fundamentals/realms.txt [hunk 12]
```diff
@@ -237,6 +237,15 @@ autorelease pool when accessing {+realm+} from a dispatch queue.
 
 .. _ios-default-realm:
 
+App Download File Size
```
Comment (cbush) [inline source/sdk/ios/fundamentals/realms.txt:240]: This section is labeled `ios-default-realm` above. Must move above the label.

**Review by cbush — CHANGES_REQUESTED** [review]
I like the general direction but I think the first section needs work.

File: source/sdk/ios/fundamentals/object-models-and-schemas.txt (hunk 7 above)
Comment (dacharyc) [reply source/sdk/ios/fundamentals/object-models-and-schemas.txt:126]: This content is on the Object Models & Schemas page. Is there another part of the page where you think this should link, @cbush ?

File: source/sdk/ios/fundamentals/realms.txt (hunk 12 above)
Comment (dacharyc) [reply source/sdk/ios/fundamentals/realms.txt:240]: Thank you for spotting that! I moved this on the page a couple of times and eventually just missed it when this became the final resting place for that content.

File: source/sdk/ios/fundamentals/object-models-and-schemas.txt [hunk 13]
```diff
@@ -113,28 +113,28 @@ Realm Does Not Support Swift Structs
...
 developers use this as a workaround for temporary limitations in our 
 library. 
 
-To make this easier, we’ve implemented standalone/detached {+service-short+} 
-objects to behave like :apple:`Apple's NSObject <documentation/objectivec/nsobject>`. 
-We support making “in-memory copies” of {+service-short+} objects. Use an initializer 
-to create a standalone object by key-value coding copying a persisted object’s properties: 
+To make this easier, we've made unmanaged objects behave like 
+:apple:`Apple's NSObject <documentation/objectivec/nsobject>`. Use an 
+initializer to create an unmanaged object by key-value coding copying a
```
Comment (cbush) [inline source/sdk/ios/fundamentals/object-models-and-schemas.txt:136]: Can we simplify this?

>Create an unmanaged copy by passing the managed instance to the object's initializer.

File: source/sdk/ios/fundamentals/object-models-and-schemas.txt (hunk 7 above)
Comment (cbush) [reply source/sdk/ios/fundamentals/object-models-and-schemas.txt:126]: Oh right. I was thinking about the intro paragraph. But there's not a specific place where we talk about the fact that you don't need an ORM, etc. for Realm. Oh well.

File: source/sdk/ios/fundamentals/realms.txt [hunk 14]
```diff
@@ -235,8 +235,6 @@ code. Realm cannot reuse intermediate versions of the data until the
 dispatch pool deallocates the {+realm+} object. Use an explicit 
 autorelease pool when accessing {+realm+} from a dispatch queue.
 
-.. _ios-default-realm:
-
 App Download File Size
```
Comment (cbush) [inline source/sdk/ios/fundamentals/realms.txt:238]: Should we add a section label?

.. _ios-app-download-file-size:

**Review by cbush — APPROVED** [review]
Much improved!

File: source/sdk/ios/fundamentals/object-models-and-schemas.txt [hunk 15]
```diff
@@ -109,6 +109,37 @@ boilerplate query code and capacity for error.
    To learn how to add relationships to your Realm object, see
    :ref:`ios-declare-relationship-properties`.
 
+Realm Does Not Support Swift Structs
```
[the remark this answers is not in the document]
Comment (dacharyc) [reply source/sdk/ios/fundamentals/object-models-and-schemas.txt]: It does. And I don't like a negative header in general. I've shortened it to just "Structs." What do you think?

File: source/sdk/ios/fundamentals/object-models-and-schemas.txt (hunk 13 above)
Comment (dacharyc) [reply source/sdk/ios/fundamentals/object-models-and-schemas.txt]: I've [rewritten this paragraph](https://github.com/mongodb/docs-realm/pull/978/commits/e5008ff2a529e0be704fc0c1ed6a9b5d8d60dd33#diff-ea0e01aa4cd7351e0e98d04b778142d1e208bf07da90437e9cd585ce922afed7R136) a bit. What do you think, @cbush ?

Comment (cbush) [reply source/sdk/ios/fundamentals/object-models-and-schemas.txt]: Perfect.

File: source/sdk/ios/fundamentals/object-models-and-schemas.txt (hunk 15 above)
[the remark this answers is not in the document]
Comment (cbush) [reply source/sdk/ios/fundamentals/object-models-and-schemas.txt]: I can't currently check the staged build due to https://jira.mongodb.org/browse/MARKETING-21151, but I think the name is fine.

[the remark this answers is not in the document]
Comment (cbush) [reply source/sdk/ios/fundamentals/object-models-and-schemas.txt]: You might want to specify "Swift Structs". Obj-C objects can't go in (plain C) structs in Obj-C, so it's kind of implied.

**Review by ericjordanmossman — APPROVED** [review]
> The example linked to in the "source code" link is the Realm Cocoa implementation of analytics, but I found stuff in several of the SDK repos related to this, so I think it applies more broadly.

Just a note:
While it's true other sdks are also collecting similar build analytics, the way the analytics is disabled might be different (and have different comments, methods for gathering, etc.). So the broader section on analytics may still benefit from linking to each SDKs analytics source file.

**Comment by ericjordanmossman** [comment]
Hmm.. just looked at some other sdks, they also just check for an env "REALM_DISABLE_ANALYTICS". Just FYI.

**Pull request merged**